src/utils/throttle.js

```diff
@@ -0,0 +1,38 @@
+import https from 'node:https'
+import throttledQueue from 'throttled-queue'
+
+let httpsAgent
+let throttle
+
+export const configure = options => {
+  if (options.throttleRequests) {
+    throttle = throttledQueue(
+      Number(options.throttleLimit) || 3,
```

Comment:
Thank you @samblacklock for your comment, the `Number` constructor provides an extra layer that the given values even if it's in form of a string will be converted to number format.

Reply:
@krish4uu makes sense, although it may be a little overkill given its restricted by the type definition? Also its not used elsewhere in the file e.g. `options.httpKeepAliveInterval`. Personally, I would opt not to use it